pkg/api/types.go

```diff
@@ -2154,3 +2154,38 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldBy"`
+	// How long the lock is valid.
+	LeaseSeconds uint64 `json:"leaseSeconds"`
```

Comment:
@bgrant0607 Internally, TTLs in the registry are uint64s:

TTLFunc func(obj runtime.Object, existing uint64, update bool) (uint64, error)

This then feeds into the ttl provided when setting an object in etcd.  We are really just exposing what etcd wants in a ttl in LeaseSeconds.

While we can convert from int64->uint64, I don't think that's ideal.  The LeaseSeconds as a uint64 provides the exact data all ttl functionality desires.  If this is a problem, them imo we should solve it as a different issue.